app/lib/create-temp-reference.js

```diff
@@ -1,16 +1,20 @@
 const { randomInt } = require('node:crypto')
+const { profanity } = require('@2toad/profanity')
 
-/**
- * Generate unique reference number, prefixed by TEMP
- * ex. TEMP-Z4F1-F2PC
- * @returns string
- */
-function generateRandomID () {
+const containsSwearWord = (input) => {
+  return profanity.exists(input)
+}
+
+const generateRandomID = () => {
   const charset = 'ABCDEFGHIJKLMNPQRSTUVWXYZ123456789'
   const id = Array.from({ length: 8 }, () => charset.charAt(randomInt(0, charset.length))).join('')
   const firstFour = id.slice(0, 4)
   const secondFour = id.slice(4)
 
+  if (containsSwearWord(firstFour) || containsSwearWord(secondFour) || containsSwearWord(`${firstFour}${secondFour}`)) {
```

Comment:
I think the third check covers the first two as well.

Reply:
Thats a very good point! Will remove the first 2